src/app/prt-staking/components/prt-section/prt-widget.tsx

```diff
@@ -51,28 +63,39 @@ export function PrtWidget({ token, onClose }: Props) {
     return ''
   }, [currentTab])
 
-  const onClickTradeButton = useCallback(() => {
+  const onClickTradeButton = useCallback(async () => {
     if (currentTab === WidgetTab.STAKE) {
-      stakePrts(parseUnits(inputAmount, token.prtTokenData.decimals))
+      if (!isApproved) {
+        await onApprove()
```

Comment:
Ensure the user has approved the token

Reply:
Should there be a `return`?

Reply:
@janndriessen not in this case - we want to continue with the other operations (stake and then refetch)